In [1]:
import pandas as pd 
import re
import numpy as np


Proceso de Extracción, Transformación y Carga (ETL)

In [6]:
Hechos_Homicidios = pd.read_excel("../Datasets/homicidios.xlsx", sheet_name="HECHOS")
Victimas_Homicidios = pd.read_excel("../Datasets/homicidios.xlsx", sheet_name="VICTIMAS")

Dataset Hechos_Homicidios

In [9]:
Hechos_Homicidios

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
691  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
692  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
693  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
694  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
695  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                           AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   
..                                         ...           ...  ...     ...   
691                          AV. RIESTRA Y MOM       AVENIDA  ...     NaN   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...     NaN   
693                        AV. GAONA Y TERRADA       AVENIDA  ...     NaN   
694                         AV. EVA PERON 4071       AVENIDA  ...  4071.0   
695                    PADRE CARLOS MUJICA 709         CALLE  ...   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
693   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
694    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
695  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    A

In [183]:
Hechos_Homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Me di cuenta de que la columna 'HH', que muestra la franja horaria completa, está como 'Object'. Así que decidí cambiarla a 'Entero'."

In [10]:
Hechos_Homicidios["HH"] = pd.to_numeric(Hechos_Homicidios ["HH"], errors="coerce")
Hechos_Homicidios["HH"] = Hechos_Homicidios["HH"].astype("Int64")

In [11]:
Hechos_Homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     695 non-null    Int64         
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Revisión de datos nulos

In [13]:
Hechos_Homicidios[Hechos_Homicidios["Altura"].isnull()][["Dirección Normalizada", "Cruce","Altura", "Calle"]]

Dirección Normalizada  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.   
5                     27 DE FEBRERO AV. y ESCALADA AV.   
..                                                 ...   
689                           BAIGORRIA y HUGO, VICTOR   
690                     RIVADAVIA AV. y PUEYRREDON AV.   
691                                  RIESTRA AV. y MOM   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.   
693                                GAONA AV. y TERRADA   

                                   Cruce  Altura                         Calle  
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.     NaN              PIEDRA BUENA AV.  
1                    DE LOS CORRALES AV.     NaN                PAZ, GRAL. AV.  
3               VILLEGAS, CONRADO, GRAL.     NaN                LARRAZABAL AV.  
4                SAENZ PE?A, LUIS, PRES.     NaN                  SAN JUAN AV.  
5                           ESCALADA AV.     NaN             27 DE FEBRERO AV.  
..                                   ...     ...                           ...  
689                         HUGO, VICTOR     NaN                     BAIGORRIA  
690                       PUEYRREDON AV.     NaN                 RIVADAVIA AV.  
691                                  MOM     NaN                   RIESTRA AV.  
692                          LACARRA AV.     NaN  DELLEPIANE, LUIS, TTE. GRAL.  
693                              TERRADA     NaN                     GAONA AV.  

[567 rows x 4 columns]

La columna 'Altura' tiene 567 espacios vacíos, un 81,5% del total. Así que decidí eliminarla porque no hay manera de llenar todos esos vacíos."

In [14]:
Hechos_Homicidios= Hechos_Homicidios.drop("Altura", axis=1)

In [15]:
Hechos_Homicidios[Hechos_Homicidios["Dirección Normalizada"].isnull()][["Dirección Normalizada", "Cruce", "Calle", "LUGAR_DEL_HECHO"]]

Dirección Normalizada Cruce                                 Calle  \
38                    NaN   NaN                 LUGONES, LEOPOLDO AV.   
106                   NaN   NaN     AUTOPISTA BUENOS AIRES - LA PLATA   
119                   NaN   NaN                                   NaN   
180                   NaN   NaN               AUTOPISTA PERITO MORENO   
181                   NaN   NaN  AUTOPISTA DELLEPIANE LUIS TTE. GRAL.   
313                   NaN   NaN                 LUGONES, LEOPOLDO AV.   
546                   NaN   NaN                 LUGONES, LEOPOLDO AV.   
621                   NaN   NaN     AUTOPISTA BUENOS AIRES - LA PLATA   

                             LUGAR_DEL_HECHO  
38                AUTOPISTA LUGONES PK 10000  
106         AU BUENOS AIRES - LA PLATA KM. 4  
119                                       SD  
180  AU PERITO MORENO Y RAMAL ENLACE AU1/AU6  
181                       AU DELLEPIANE 2400  
313                 AUTOPISTA LUGONES KM 4.7  
546             LUGONES, LEOPOLDO AV. KM 6,1  
621         AU BUENOS AIRES LA PLATA  KM 4,5

 La columna 'Dirección Normalizada' se compone de los datos de 'Cruce' y 'Calle'. Como los datos de 'Cruce' también están faltantes, no hay manera de armar correctamente la dirección. Pero, solo hay 8 valores perdidos en 'Dirección Normalizada', lo que representa solo el 1,1% del total. Así que, por ahora, prefiero conservarla.

In [16]:
Hechos_Homicidios[Hechos_Homicidios["Cruce"].isnull()][["Dirección Normalizada", "Cruce", "Calle"]]


Dirección Normalizada Cruce  \
2                           ENTRE RIOS AV. 2034   NaN   
9                           ENTRE RIOS AV. 1366   NaN   
14                                 SUIPACHA 156   NaN   
33                                    LIMA 1483   NaN   
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI   NaN   
..                                          ...   ...   
680                AUTOPISTA 25 DE MAYO y PASCO   NaN   
685                              ESMERALDA 1359   NaN   
687                              GAONA AV. 3655   NaN   
694                         PERON, EVA AV. 4071   NaN   
695                     PADRE CARLOS MUGICA 709   NaN   

                                          Calle  
2                                ENTRE RIOS AV.  
9                                ENTRE RIOS AV.  
14                                     SUIPACHA  
33                                         LIMA  
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI  
..                                          ...  
680                        AUTOPISTA 25 DE MAYO  
685                                   ESMERALDA  
687                                   GAONA AV.  
694                              PERON, EVA AV.  
695                         PADRE CARLOS MUJICA  

[171 rows x 3 columns]

En la fila 680, falta el dato en la columna 'Cruce', así que procedo a rellenarlo

In [17]:
fila = 680
columna = 'Cruce'

In [18]:
Hechos_Homicidios.at[fila, columna] = "PASCO"

In [19]:
Hechos_Homicidios[Hechos_Homicidios["Cruce"].isnull()][["Dirección Normalizada", "Cruce", "Calle"]]

Dirección Normalizada Cruce  \
2                                  ENTRE RIOS AV. 2034   NaN   
9                                  ENTRE RIOS AV. 1366   NaN   
14                                        SUIPACHA 156   NaN   
33                                           LIMA 1483   NaN   
35          AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI   NaN   
..                                                 ...   ...   
679  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI y B...   NaN   
685                                     ESMERALDA 1359   NaN   
687                                     GAONA AV. 3655   NaN   
694                                PERON, EVA AV. 4071   NaN   
695                            PADRE CARLOS MUGICA 709   NaN   

                                          Calle  
2                                ENTRE RIOS AV.  
9                                ENTRE RIOS AV.  
14                                     SUIPACHA  
33                                         LIMA  
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI  
..                                          ...  
679  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI  
685                                   ESMERALDA  
687                                   GAONA AV.  
694                              PERON, EVA AV.  
695                         PADRE CARLOS MUJICA  

[170 rows x 3 columns]

In [193]:
Hechos_Homicidios.iloc[679] 

ID                                                               2021-0081
N_VICTIMAS                                                               1
FECHA                                                  2021-11-02 00:00:00
AAAA                                                                  2021
MM                                                                      11
DD                                                                       2
HORA                                                              13:35:00
HH                                                                      13
LUGAR_DEL_HECHO                                AU FRONDIZI SUBIDA BRANDSEN
TIPO_DE_CALLE                                                    AUTOPISTA
Calle                           AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI
Cruce                                                                  NaN
Dirección Normalizada    AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI y B...
COMUNA                   

En la fila 679 se encuentra el dato faltante de la Columna Cruce y procedo a completarlo

In [20]:
fila1 = 679
columna1 = 'Cruce'

In [22]:
Hechos_Homicidios.at[fila1, columna1] = "BRANDSEN"

In [23]:
Hechos_Homicidios[Hechos_Homicidios["Cruce"].isnull()][["Dirección Normalizada", "Cruce", "Calle"]]

Dirección Normalizada Cruce  \
2                           ENTRE RIOS AV. 2034   NaN   
9                           ENTRE RIOS AV. 1366   NaN   
14                                 SUIPACHA 156   NaN   
33                                    LIMA 1483   NaN   
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI   NaN   
..                                          ...   ...   
677                                TUCUMAN 3112   NaN   
685                              ESMERALDA 1359   NaN   
687                              GAONA AV. 3655   NaN   
694                         PERON, EVA AV. 4071   NaN   
695                     PADRE CARLOS MUGICA 709   NaN   

                                          Calle  
2                                ENTRE RIOS AV.  
9                                ENTRE RIOS AV.  
14                                     SUIPACHA  
33                                         LIMA  
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI  
..                                          ...  
677                                     TUCUMAN  
685                                   ESMERALDA  
687                                   GAONA AV.  
694                              PERON, EVA AV.  
695                         PADRE CARLOS MUJICA  

[169 rows x 3 columns]

La columna 'Cruce' tiene 169 datos perdidos, lo que es como el 24,28% del total. Pero voy a mantenerla de todos modos

In [24]:
Hechos_Homicidios[Hechos_Homicidios["Calle"].isnull()][["Dirección Normalizada", "Cruce", "Calle", "LUGAR_DEL_HECHO"]]

Dirección Normalizada Cruce Calle LUGAR_DEL_HECHO
119                   NaN   NaN   NaN              SD

La columna Calle tiene un solo dato nulo, que no se puede completar con datos de otras columnas porque también se encuentran nulas. 

Verificación de datos duplicados

In [25]:
Hechos_Homicidios.duplicated().sum()

0

No se encontraron datos duplicados

Desarrollo una función para extraer las coordenadas de la columna 'XY (CABA)'. Esta función utiliza expresiones regulares para identificar todos los números decimales presentes en la columna. Posteriormente, los valores encontrados se almacenan en una lista denominada 'coordenadas', la cual se aplica a la columna mediante el método '.map()'."

In [26]:
def extraer_coordenadas(texto):
    coordenadas = re.findall(r'\d+\.\d+', texto)
    if len(coordenadas) == 2:
        return float(coordenadas[0]), float(coordenadas[1])
    else:
        return None, None

# Se aplica la función a la columna 'XY (CABA)'
Hechos_Homicidios['Coordenada_X'], Hechos_Homicidios['Coordenada_Y'] = zip(*Hechos_Homicidios['XY (CABA)'].map(extraer_coordenadas))

# se filtran los registros que no tienen coordenadas válidas
Hechos_Homicidios = Hechos_Homicidios[(Hechos_Homicidios['Coordenada_X'].notnull()) & (Hechos_Homicidios['Coordenada_Y'].notnull())]

In [27]:
Hechos_Homicidios

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
691  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
692  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
693  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
694  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
695  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...   
2                           AV ENTRE RIOS 2034       AVENIDA  ...   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...   
..                                         ...           ...  ...   
691                          AV. RIESTRA Y MOM       AVENIDA  ...   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...   
693                        AV. GAONA Y TERRADA       AVENIDA  ...   
694                         AV. EVA PERON 4071       AVENIDA  ...   
695                    PADRE CARLOS MUJICA 709         CALLE  ...   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
693   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
694    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
695  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    ACUSADO   Coordenada_X   Coordenada_Y  
0         MOTO-AUTO       MOTO       AUTO   98896.782384   93532.434378  
1    AUTO-PASAJEROS       AUTO  PASAJEROS   95832.055711   95505.416420  
2         MOTO-AUTO       MOTO       AUTO  106684.290900   99706.576878  
3           MOTO-SD       MOTO         SD   99840.652248   94269.165344  
4    MOTO-PASAJEROS       MOTO  PASAJEROS  106980.328279  100752.169158  
..              ...        ...        ...            ...            ...  
691       MOTO-AUTO       MOTO       AUTO  102728.600901   98186.249292  
692       MOTO-AUTO       MOTO       AUTO   99624.297958   97569.698

Se encontró en la página del Gobierno de la Ciudad de Buenos Aires un dataset con los numeros de las Comunas junto con los barrios que las integran. Se procedió a descargarlo para agregar en el DF Homicidios una columna con los barrios que integran las comunas.

In [28]:
Comunas=pd.read_excel("../Datasets/comunas.xlsx")

In [29]:
Comunas

COMUNA                                            BARRIOS
0        1  CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...
1        2                                           RECOLETA
2        3                          BALVANERA - SAN CRISTOBAL
3        4  BARRACAS - BOCA - NUEVA POMPEYA - PARQUE PATRI...
4        5                                    ALMAGRO - BOEDO
5        6                                          CABALLITO
6        7                          FLORES - PARQUE CHACABUCO
7        8     VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI
8        9            LINIERS - MATADEROS - PARQUE AVELLANEDA
9       10  FLORESTA - MONTE CASTRO - VELEZ SARSFIELD - VE...
10      11  VILLA DEL PARQUE - VILLA  DEVOTO - VILLA GRAL....
11      12  COGHLAN - SAAVEDRA - VILLA PUEYRREDON - VILLA ...
12      13                     BELGRANO - COLEGIALES - NUÑEZ
13      14                                            PALERMO
14      15  AGRONOMIA - CHACARITA -  PARQUE CHAS -  PATERN...

Realizo un merge entre los DataFrames de Homicidios y Comunas, agregando solo la columna 'Barrios' al conjunto de datos combinado.

In [30]:
Hechos_Homicidios=Hechos_Homicidios.merge(Comunas, on="COMUNA", how="left")

In [31]:
Hechos_Homicidios

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
677  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
678  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
679  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
680  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
681  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ... COMUNA  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...      8   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...      9   
2                           AV ENTRE RIOS 2034       AVENIDA  ...      1   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...      8   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...      1   
..                                         ...           ...  ...    ...   
677                          AV. RIESTRA Y MOM       AVENIDA  ...      7   
678                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...      9   
679                        AV. GAONA Y TERRADA       AVENIDA  ...     11   
680                         AV. EVA PERON 4071       AVENIDA  ...      9   
681                    PADRE CARLOS MUJICA 709         CALLE  ...      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
677   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
678    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
679   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
680    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
681  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    ACUSADO   Coordenada_X   Coordenada_Y  \
0         MOTO-AUTO       MOTO       AUTO   98896.782384   93532.434378   
1    AUTO-PASAJEROS       AUTO  PASAJEROS   95832.055711   95505.416420   
2         MOTO-AUTO       MOTO       AUTO  106684.290900   99706.576878   
3           MOTO-SD       MOTO         SD   99840.652248   94269.165344   
4    MOTO-PASAJEROS       MOTO  PASAJEROS  106980.328279  100752.169158   
..              ...        ...        ...            ...            ...   
677       MOTO-AUTO       MOTO       AUTO  102728.600901   98186.249292   
678       MOTO-AUTO       MOTO       AUTO   99624.297958   97569.698011   
679     MOTO-CARGAS       MOTO     CARGAS   99116.454924  101045.232848   
680     AUTO-CARGAS       AUTO     CARGAS   99324.544640   97676.269324   
681  BICICLETA-AUTO  BICICLETA       AUTO  107664.166478  104708.639621   

                                               BARRIOS  
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI  
1              LINIERS - MATADEROS - PARQUE AVELLANEDA  
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI  
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  
..                                                 ...  
677                          FLORES - PARQUE CHACABUCO  


Se capitalizan los nombres de las columnas para normalizarlas

In [32]:
Hechos_Homicidios.columns = Hechos_Homicidios.columns.str.capitalize()

In [33]:
Hechos_Homicidios

Id  N_victimas      Fecha  Aaaa  Mm  Dd      Hora  Hh  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
677  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
678  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
679  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
680  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
681  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               Lugar_del_hecho Tipo_de_calle  ... Comuna  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...      8   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...      9   
2                           AV ENTRE RIOS 2034       AVENIDA  ...      1   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...      8   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...      1   
..                                         ...           ...  ...    ...   
677                          AV. RIESTRA Y MOM       AVENIDA  ...      7   
678                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...      9   
679                        AV. GAONA Y TERRADA       AVENIDA  ...     11   
680                         AV. EVA PERON 4071       AVENIDA  ...      9   
681                    PADRE CARLOS MUJICA 709         CALLE  ...      1   

                                   Xy (caba)         Pos x         Pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
677   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
678    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
679   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
680    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
681  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      Participantes    Victima    Acusado   Coordenada_x   Coordenada_y  \
0         MOTO-AUTO       MOTO       AUTO   98896.782384   93532.434378   
1    AUTO-PASAJEROS       AUTO  PASAJEROS   95832.055711   95505.416420   
2         MOTO-AUTO       MOTO       AUTO  106684.290900   99706.576878   
3           MOTO-SD       MOTO         SD   99840.652248   94269.165344   
4    MOTO-PASAJEROS       MOTO  PASAJEROS  106980.328279  100752.169158   
..              ...        ...        ...            ...            ...   
677       MOTO-AUTO       MOTO       AUTO  102728.600901   98186.249292   
678       MOTO-AUTO       MOTO       AUTO   99624.297958   97569.698011   
679     MOTO-CARGAS       MOTO     CARGAS   99116.454924  101045.232848   
680     AUTO-CARGAS       AUTO     CARGAS   99324.544640   97676.269324   
681  BICICLETA-AUTO  BICICLETA       AUTO  107664.166478  104708.639621   

                                               Barrios  
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI  
1              LINIERS - MATADEROS - PARQUE AVELLANEDA  
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI  
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  
..                                                 ...  
677                          FLORES - PARQUE CHACABUCO  


Elimino la columna 'XY (CABA)' ya que ahora cuento con las columnas 'Coordenadas_X' y 'Coordenadas_Y', las cuales tienen los mismos datos que tenía la columna eliminada

In [34]:
Hechos_Homicidios= Hechos_Homicidios.drop("Xy (caba)", axis=1)

Procedo a renombrar las siguientes columnas a modo de buena práctica

In [35]:
Hechos_Homicidios= Hechos_Homicidios.rename(columns={"N_victimas": "Numero_víctimas",
                                            "Aaaa": "Año",
                                            "Mm":"Mes",
                                            "Dd": "Día",
                                            "Hora": "Hora completa",
                                            "Hh": "Hora"}) 

 Se decide crear la columna Día_Semana donde se corresponde cada fecha con un dia de la semana para poder analizarlos luego con los siniestros en cuestión.

In [36]:
Hechos_Homicidios['Día_Semana'] = Hechos_Homicidios['Fecha'].dt.strftime('%A')

#Creo un diccionario para pasar los nombres de los días de la semana al español y lo aplico a la columna nueva
dic={  'Friday':'Viernes',
       'Saturday': 'Sabado',
       'Sunday': 'Domingo',  
       'Thursday': 'Jueves',
        'Monday': 'Lunes',
        'Wednesday': 'Miercoles',
       'Tuesday': 'Martes'}
Hechos_Homicidios['Día_Semana'] =Hechos_Homicidios['Día_Semana'].replace(dic)

In [37]:
Hechos_Homicidios

Id  Numero_víctimas      Fecha   Año  Mes  Día Hora completa  \
0    2016-0001                1 2016-01-01  2016    1    1      04:00:00   
1    2016-0002                1 2016-01-02  2016    1    2      01:15:00   
2    2016-0003                1 2016-01-03  2016    1    3      07:00:00   
3    2016-0004                1 2016-01-10  2016    1   10      00:00:00   
4    2016-0005                1 2016-01-21  2016    1   21      05:20:00   
..         ...              ...        ...   ...  ...  ...           ...   
677  2021-0093                1 2021-12-13  2021   12   13      17:10:00   
678  2021-0094                1 2021-12-20  2021   12   20      01:10:00   
679  2021-0095                1 2021-12-30  2021   12   30      00:43:00   
680  2021-0096                1 2021-12-15  2021   12   15      10:30:00   
681  2021-0097                1 2021-11-18  2021   11   18      06:10:00   

     Hora                            Lugar_del_hecho Tipo_de_calle  ...  \
0       4  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...   
1       1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...   
2       7                         AV ENTRE RIOS 2034       AVENIDA  ...   
3       0      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...   
4       5  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...   
..    ...                                        ...           ...  ...   
677    17                          AV. RIESTRA Y MOM       AVENIDA  ...   
678     1                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...   
679     0                        AV. GAONA Y TERRADA       AVENIDA  ...   
680    10                         AV. EVA PERON 4071       AVENIDA  ...   
681     6                    PADRE CARLOS MUJICA 709         CALLE  ...   

    Comuna         Pos x         Pos y   Participantes    Victima    Acusado  \
0        8  -58.47533969  -34.68757022       MOTO-AUTO       MOTO       AUTO   
1        9  -58.50877521  -34.66977709  AUTO-PASAJEROS       AUTO  PASAJEROS   
2        1  -58.39040293  -34.63189362       MOTO-AUTO       MOTO       AUTO   
3        8  -58.46503904  -34.68092974         MOTO-SD       MOTO         SD   
4        1  -58.38718297  -34.62246630  MOTO-PASAJEROS       MOTO  PASAJEROS   
..     ...           ...           ...             ...        ...        ...   
677      7  -58.43353773  -34.64561636       MOTO-AUTO       MOTO       AUTO   
678      9  -58.46739825  -34.65117757       MOTO-AUTO       MOTO       AUTO   
679     11  -58.47293407  -34.61984745     MOTO-CARGAS       MOTO     CARGAS   
680      9  -58.47066794  -34.65021673     AUTO-CARGAS       AUTO     CARGAS   
681      1  -58.37976155  -34.58679619  BICICLETA-AUTO  BICICLETA       AUTO   

      Coordenada_x   Coordenada_y  \
0     98896.782384   93532.434378   
1     95832.055711   95505.416420   
2    106684.290900   99706.576878   
3     99840.652248   94269.165344   
4    106980.328279  100752.169158   
..             ...            ...   
677  102728.600901   98186.249292   
678   99624.297958   97569.698011   
679   99116.454924  101045.232848   
680   99324.544640   97676.269324   
681  107664.166478  104708.639621   

                                               Barrios  Día_Semana  
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI     Viernes  
1              LINIERS - MATADEROS - PARQUE AVELLANEDA      Sabado  
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...     Domingo  
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI     Domingo  
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...      Jueves  
..                                                 ...         ...  
677                          FLORES - PARQUE CHACABUCO       Lunes  
678            LINIERS - MATADEROS - PARQUE AVELLANEDA       Lunes  
679  VILLA DEL PARQUE - VILLA  DEVOTO - VILLA GRAL....      Jueves  
680            LINIERS - MATADEROS - PARQUE AVELLANEDA   Miercoles  
681  CONSTITUCION - MONTSERRAT - 

Dataset Victimas_Homicidios

In [38]:
Victimas_Homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID_hecho             717 non-null    object        
 1   FECHA                717 non-null    datetime64[ns]
 2   AAAA                 717 non-null    int64         
 3   MM                   717 non-null    int64         
 4   DD                   717 non-null    int64         
 5   ROL                  717 non-null    object        
 6   VICTIMA              717 non-null    object        
 7   SEXO                 717 non-null    object        
 8   EDAD                 717 non-null    object        
 9   FECHA_FALLECIMIENTO  717 non-null    object        
dtypes: datetime64[ns](1), int64(3), object(6)
memory usage: 56.1+ KB


Se puede observar que ninguna columna posee datos nulos.

Modifico el tipo de dato de la columna Edad (Object) por tipo de dato Entero

In [39]:
Victimas_Homicidios ["EDAD"] = pd.to_numeric(Victimas_Homicidios["EDAD"], errors="coerce")
Victimas_Homicidios ["EDAD"] = Victimas_Homicidios["EDAD"].astype("Int64")

Cambio el tipo de dato de la columna FECHA_FALLECIMIENTO (Object) por tipo de dato Datetime pero solo en formato fecha, sin la hora

In [213]:
Victimas_Homicidios["FECHA_FALLECIMIENTO"] = pd.to_datetime(Victimas_Homicidios["FECHA_FALLECIMIENTO"], format='%Y-%m-%d', errors='coerce')

Victimas_Homicidios["FECHA_FALLECIMIENTO"] = Victimas_Homicidios["FECHA_FALLECIMIENTO"].dt.date

Creo la columna Rango_Etario basada en la columna Edad, utilizando la función `pd.cut()`, para posteriormente analizar de manera eficiente los datos 

In [40]:
limites = [0, 20, 40, 60, 80, 100] # Acá se establecen los límites de los rangos etarios y las etiquetas
etiquetas = ['Menos de 20', ' Entre 21- 40', 'Entre 41-60', 'Entre 61-80', 'Más de 80'] # establecen las etiquetas

# Usar pd.cut() para crear la columna de rango etario
Victimas_Homicidios['Rango_Etario'] = pd.cut(Victimas_Homicidios['EDAD'], bins=limites, labels=etiquetas, right=False)

In [41]:
Victimas_Homicidios

ID_hecho      FECHA  AAAA  MM  DD                   ROL VICTIMA  \
0    2016-0001 2016-01-01  2016   1   1             CONDUCTOR    MOTO   
1    2016-0002 2016-01-02  2016   1   2             CONDUCTOR    AUTO   
2    2016-0003 2016-01-03  2016   1   3             CONDUCTOR    MOTO   
3    2016-0004 2016-01-10  2016   1  10             CONDUCTOR    MOTO   
4    2016-0005 2016-01-21  2016   1  21             CONDUCTOR    MOTO   
..         ...        ...   ...  ..  ..                   ...     ...   
712  2021-0092 2021-12-12  2021  12  12                PEATON  PEATON   
713  2021-0093 2021-12-13  2021  12  13  PASAJERO_ACOMPAÑANTE    MOTO   
714  2021-0094 2021-12-20  2021  12  20  PASAJERO_ACOMPAÑANTE    MOTO   
715  2021-0095 2021-12-30  2021  12  30             CONDUCTOR    MOTO   
716  2021-0096 2021-12-15  2021  12  15             CONDUCTOR    AUTO   

          SEXO  EDAD  FECHA_FALLECIMIENTO   Rango_Etario  
0    MASCULINO    19  2016-01-01 00:00:00    Menos de 20  
1    MASCULINO    70  2016-01-02 00:00:00    Entre 61-80  
2    MASCULINO    30  2016-01-03 00:00:00   Entre 21- 40  
3    MASCULINO    18                   SD    Menos de 20  
4    MASCULINO    29  2016-02-01 00:00:00   Entre 21- 40  
..         ...   ...                  ...            ...  
712   FEMENINO    50  2021-12-12 00:00:00    Entre 41-60  
713   FEMENINO    18  2021-12-18 00:00:00    Menos de 20  
714   FEMENINO    43  2021-12-20 00:00:00    Entre 41-60  
715  MASCULINO    27  2022-01-02 00:00:00   Entre 21- 40  
716  MASCULINO    60  2021-12-20 00:00:00    Entre 61-80  

[717 rows x 11 columns]

Se capitalizan los nombres de las columnas para normalizarlas

In [42]:
Victimas_Homicidios.columns = Victimas_Homicidios.columns.str.capitalize()

Procedo a renombrar las siguientes columnas para mejor entendimiento

In [44]:
Victimas_Homicidios= Victimas_Homicidios.rename(columns={'Id_hecho': 'Id',
                                                      'Aaaa':'Año',
                                                      'Mm':'Mes',
                                                      'Dd':'Día',
                                                      'Victima':'Víctima'}) 

In [45]:
Victimas_Homicidios

Id      Fecha   Año  Mes  Día                   Rol Víctima  \
0    2016-0001 2016-01-01  2016    1    1             CONDUCTOR    MOTO   
1    2016-0002 2016-01-02  2016    1    2             CONDUCTOR    AUTO   
2    2016-0003 2016-01-03  2016    1    3             CONDUCTOR    MOTO   
3    2016-0004 2016-01-10  2016    1   10             CONDUCTOR    MOTO   
4    2016-0005 2016-01-21  2016    1   21             CONDUCTOR    MOTO   
..         ...        ...   ...  ...  ...                   ...     ...   
712  2021-0092 2021-12-12  2021   12   12                PEATON  PEATON   
713  2021-0093 2021-12-13  2021   12   13  PASAJERO_ACOMPAÑANTE    MOTO   
714  2021-0094 2021-12-20  2021   12   20  PASAJERO_ACOMPAÑANTE    MOTO   
715  2021-0095 2021-12-30  2021   12   30             CONDUCTOR    MOTO   
716  2021-0096 2021-12-15  2021   12   15             CONDUCTOR    AUTO   

          Sexo  Edad  Fecha_fallecimiento   Rango_etario  
0    MASCULINO    19  2016-01-01 00:00:00    Menos de 20  
1    MASCULINO    70  2016-01-02 00:00:00    Entre 61-80  
2    MASCULINO    30  2016-01-03 00:00:00   Entre 21- 40  
3    MASCULINO    18                   SD    Menos de 20  
4    MASCULINO    29  2016-02-01 00:00:00   Entre 21- 40  
..         ...   ...                  ...            ...  
712   FEMENINO    50  2021-12-12 00:00:00    Entre 41-60  
713   FEMENINO    18  2021-12-18 00:00:00    Menos de 20  
714   FEMENINO    43  2021-12-20 00:00:00    Entre 41-60  
715  MASCULINO    27  2022-01-02 00:00:00   Entre 21- 40  
716  MASCULINO    60  2021-12-20 00:00:00    Entre 61-80  

[717 rows x 11 columns]

Verificación de si ambos datasets tienen información compartida

In [46]:
Victimas_Homicidios[Victimas_Homicidios["Id"]=="2016-0001"]


Id      Fecha   Año  Mes  Día        Rol Víctima       Sexo  Edad  \
0  2016-0001 2016-01-01  2016    1    1  CONDUCTOR    MOTO  MASCULINO    19   

   Fecha_fallecimiento Rango_etario  
0  2016-01-01 00:00:00  Menos de 20

In [47]:
Hechos_Homicidios[Hechos_Homicidios["Id"]=="2016-0001"]

Id  Numero_víctimas      Fecha   Año  Mes  Día Hora completa  Hora  \
0  2016-0001                1 2016-01-01  2016    1    1      04:00:00     4   

                             Lugar_del_hecho Tipo_de_calle  ... Comuna  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...      8   

          Pos x         Pos y  Participantes Victima Acusado  Coordenada_x  \
0  -58.47533969  -34.68757022      MOTO-AUTO    MOTO    AUTO  98896.782384   

   Coordenada_y                                         Barrios  Día_Semana  
0  93532.434378  VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI     Viernes  

[1 rows x 23 columns]

Elimino las columnas delDf Victimas_Homicidios que se repiten en el Df Hechos_Homicidios

In [48]:
Victimas_Homicidios = Victimas_Homicidios.drop(['Fecha', 'Año', 'Mes', 'Día', 'Víctima'], axis=1)

In [49]:
Victimas_Homicidios

Id                   Rol       Sexo  Edad  Fecha_fallecimiento  \
0    2016-0001             CONDUCTOR  MASCULINO    19  2016-01-01 00:00:00   
1    2016-0002             CONDUCTOR  MASCULINO    70  2016-01-02 00:00:00   
2    2016-0003             CONDUCTOR  MASCULINO    30  2016-01-03 00:00:00   
3    2016-0004             CONDUCTOR  MASCULINO    18                   SD   
4    2016-0005             CONDUCTOR  MASCULINO    29  2016-02-01 00:00:00   
..         ...                   ...        ...   ...                  ...   
712  2021-0092                PEATON   FEMENINO    50  2021-12-12 00:00:00   
713  2021-0093  PASAJERO_ACOMPAÑANTE   FEMENINO    18  2021-12-18 00:00:00   
714  2021-0094  PASAJERO_ACOMPAÑANTE   FEMENINO    43  2021-12-20 00:00:00   
715  2021-0095             CONDUCTOR  MASCULINO    27  2022-01-02 00:00:00   
716  2021-0096             CONDUCTOR  MASCULINO    60  2021-12-20 00:00:00   

      Rango_etario  
0      Menos de 20  
1      Entre 61-80  
2     Entre 21- 40  
3      Menos de 20  
4     Entre 21- 40  
..             ...  
712    Entre 41-60  
713    Menos de 20  
714    Entre 41-60  
715   Entre 21- 40  
716    Entre 61-80  

[717 rows x 6 columns]

Reemplacé las fechas que contenían 'Nat' en la columna 'Fecha_fallecimiento' con espacios vacíos para poder manejarlas correctamente.

In [50]:
Victimas_Homicidios["Fecha_fallecimiento"] = Victimas_Homicidios ["Fecha_fallecimiento"].replace(pd.NaT,"")

In [51]:
Victimas_Homicidios

Id                   Rol       Sexo  Edad  Fecha_fallecimiento  \
0    2016-0001             CONDUCTOR  MASCULINO    19  2016-01-01 00:00:00   
1    2016-0002             CONDUCTOR  MASCULINO    70  2016-01-02 00:00:00   
2    2016-0003             CONDUCTOR  MASCULINO    30  2016-01-03 00:00:00   
3    2016-0004             CONDUCTOR  MASCULINO    18                   SD   
4    2016-0005             CONDUCTOR  MASCULINO    29  2016-02-01 00:00:00   
..         ...                   ...        ...   ...                  ...   
712  2021-0092                PEATON   FEMENINO    50  2021-12-12 00:00:00   
713  2021-0093  PASAJERO_ACOMPAÑANTE   FEMENINO    18  2021-12-18 00:00:00   
714  2021-0094  PASAJERO_ACOMPAÑANTE   FEMENINO    43  2021-12-20 00:00:00   
715  2021-0095             CONDUCTOR  MASCULINO    27  2022-01-02 00:00:00   
716  2021-0096             CONDUCTOR  MASCULINO    60  2021-12-20 00:00:00   

      Rango_etario  
0      Menos de 20  
1      Entre 61-80  
2     Entre 21- 40  
3      Menos de 20  
4     Entre 21- 40  
..             ...  
712    Entre 41-60  
713    Menos de 20  
714    Entre 41-60  
715   Entre 21- 40  
716    Entre 61-80  

[717 rows x 6 columns]

Verifico si la columna Edad posee datos "Nat"

In [52]:
Victimas_Homicidios['Edad'].unique()

<IntegerArray>
[  19,   70,   30,   18,   29,   22,   16,   59,   65,   34,   41,   50,   38,
   21,   52,   36,   20,   54, <NA>,   56,   24,   78,   79,   26,   57,   37,
   58,   23,   60,   42,   53,   51,   40,   87,   76,   75,   35,   80,   43,
   45,   67,   27,   55,   49,   81,   25,   33,   46,   83,   39,   28,    7,
   48,    4,   82,   32,   17,   47,   61,   10,   95,   73,   84,   66,   85,
    1,   15,   13,   77,   44,   31,   62,   74,   71,   11,   86,   69,   72,
   12,   63,   92,   68,   91,   64,    5,   88]
Length: 86, dtype: Int64

Sustituí los valores 'Nat' en la columna 'Edad' por espacios vacíos para poder manipularlos adecuadamente.

In [53]:
Victimas_Homicidios["Edad"] = Victimas_Homicidios["Edad"].replace([" ","SD"], np.nan)


Se unifican los datasets Victimas_Homicidios y Hechos_Homicidios utilizando la función merge 

In [54]:
Siniestro_viales= Victimas_Homicidios.merge(Hechos_Homicidios, on="Id", how="left")

In [56]:
Siniestro_viales

Id                   Rol       Sexo  Edad  Fecha_fallecimiento  \
0    2016-0001             CONDUCTOR  MASCULINO    19  2016-01-01 00:00:00   
1    2016-0002             CONDUCTOR  MASCULINO    70  2016-01-02 00:00:00   
2    2016-0003             CONDUCTOR  MASCULINO    30  2016-01-03 00:00:00   
3    2016-0004             CONDUCTOR  MASCULINO    18                   SD   
4    2016-0005             CONDUCTOR  MASCULINO    29  2016-02-01 00:00:00   
..         ...                   ...        ...   ...                  ...   
712  2021-0092                PEATON   FEMENINO    50  2021-12-12 00:00:00   
713  2021-0093  PASAJERO_ACOMPAÑANTE   FEMENINO    18  2021-12-18 00:00:00   
714  2021-0094  PASAJERO_ACOMPAÑANTE   FEMENINO    43  2021-12-20 00:00:00   
715  2021-0095             CONDUCTOR  MASCULINO    27  2022-01-02 00:00:00   
716  2021-0096             CONDUCTOR  MASCULINO    60  2021-12-20 00:00:00   

      Rango_etario  Numero_víctimas      Fecha     Año   Mes  ...  Comuna  \
0      Menos de 20              1.0 2016-01-01  2016.0   1.0  ...     8.0   
1      Entre 61-80              1.0 2016-01-02  2016.0   1.0  ...     9.0   
2     Entre 21- 40              1.0 2016-01-03  2016.0   1.0  ...     1.0   
3      Menos de 20              1.0 2016-01-10  2016.0   1.0  ...     8.0   
4     Entre 21- 40              1.0 2016-01-21  2016.0   1.0  ...     1.0   
..             ...              ...        ...     ...   ...  ...     ...   
712    Entre 41-60              1.0 2021-12-12  2021.0  12.0  ...     3.0   
713    Menos de 20              1.0 2021-12-13  2021.0  12.0  ...     7.0   
714    Entre 41-60              1.0 2021-12-20  2021.0  12.0  ...     9.0   
715   Entre 21- 40              1.0 2021-12-30  2021.0  12.0  ...    11.0   
716    Entre 61-80              1.0 2021-12-15  2021.0  12.0  ...     9.0   

            Pos x         Pos y   Participantes Victima    Acusado  \
0    -58.47533969  -34.68757022       MOTO-AUTO    MOTO       AUTO   
1    -58.50877521  -34.66977709  AUTO-PASAJEROS    AUTO  PASAJEROS   
2    -58.39040293  -34.63189362       MOTO-AUTO    MOTO       AUTO   
3    -58.46503904  -34.68092974         MOTO-SD    MOTO         SD   
4    -58.38718297  -34.62246630  MOTO-PASAJEROS    MOTO  PASAJEROS   
..            ...           ...             ...     ...        ...   
712  -58.40596860  -34.61011987     PEATON-AUTO  PEATON       AUTO   
713  -58.43353773  -34.64561636       MOTO-AUTO    MOTO       AUTO   
714  -58.46739825  -34.65117757       MOTO-AUTO    MOTO       AUTO   
715  -58.47293407  -34.61984745     MOTO-CARGAS    MOTO     CARGAS   
716  -58.47066794  -34.65021673     AUTO-CARGAS    AUTO     CARGAS   

      Coordenada_x   Coordenada_y  \
0     98896.782384   93532.434378   
1     95832.055711   95505.416420   
2    106684.290900   99706.576878   
3     99840.652248   94269.165344   
4    106980.328279  100752.169158   
..             ...            ...   
712  105258.353686  102122.932314   
713  102728.600901   98186.249292   
714   99624.297958   97569.698011   
715   99116.454924  101045.232848   
716   99324.544640   97676.269324   

                                               Barrios Día_Semana  
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI    Viernes  
1              LINIERS - MATADEROS - PARQUE AVELLANEDA     Sabado  
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...    Domingo  
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI    Domingo  
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...     Jueves  
..                                                 ...        ...  
712                          BALVANERA - SAN CRISTOBAL    Domingo  
713                          FLORES - PARQUE CHACABUCO      Lunes  
714            LINIERS - MATADEROS - PARQUE AVELLANEDA      Lunes  
715  VILLA DEL PARQUE - VILLA  DEVOTO - VILLA GRAL....     Jueves  
716            LINIERS - MATADEROS - PARQUE AVELLANEDA  Miercoles  

[717 rows x 28 columns]

In [57]:
Siniestro_viales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Id                     717 non-null    object        
 1   Rol                    717 non-null    object        
 2   Sexo                   717 non-null    object        
 3   Edad                   664 non-null    Int64         
 4   Fecha_fallecimiento    717 non-null    object        
 5   Rango_etario           664 non-null    category      
 6   Numero_víctimas        702 non-null    float64       
 7   Fecha                  702 non-null    datetime64[ns]
 8   Año                    702 non-null    float64       
 9   Mes                    702 non-null    float64       
 10  Día                    702 non-null    float64       
 11  Hora completa          702 non-null    object        
 12  Hora                   701 non-null    Int64         
 13  Lugar

Observo que en algunas columnas se modificó el tipo de dato a Flotante. Se cambiarán nuevamente a tipo de dato Entero debido a que es esencial para su manipulación

In [58]:
Siniestro_viales["Numero_víctimas"] = Siniestro_viales["Numero_víctimas"].fillna(0.0)
Siniestro_viales["Numero_víctimas"] = Siniestro_viales["Numero_víctimas"].astype ("Int64")

In [59]:
Siniestro_viales["Año"] = Siniestro_viales["Año"].astype ("Int64")
Siniestro_viales["Año"] = Siniestro_viales["Año"].replace([" ","SD"], np.nan)


In [60]:
Siniestro_viales["Mes"] = Siniestro_viales["Mes"].fillna(0.0)
Siniestro_viales["Mes"] = Siniestro_viales["Mes"].astype ("Int64")

In [61]:
Siniestro_viales["Día"] = Siniestro_viales["Día"].fillna(0.0)
Siniestro_viales["Día"] = Siniestro_viales["Día"].astype ("Int64")

In [62]:
Siniestro_viales["Comuna"] = Siniestro_viales["Comuna"].fillna(0.0)
Siniestro_viales["Comuna"] = Siniestro_viales["Comuna"].astype ("Int64")

In [63]:
Siniestro_viales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Id                     717 non-null    object        
 1   Rol                    717 non-null    object        
 2   Sexo                   717 non-null    object        
 3   Edad                   664 non-null    Int64         
 4   Fecha_fallecimiento    717 non-null    object        
 5   Rango_etario           664 non-null    category      
 6   Numero_víctimas        717 non-null    Int64         
 7   Fecha                  702 non-null    datetime64[ns]
 8   Año                    702 non-null    Int64         
 9   Mes                    717 non-null    Int64         
 10  Día                    717 non-null    Int64         
 11  Hora completa          702 non-null    object        
 12  Hora                   701 non-null    Int64         
 13  Lugar

In [64]:
Siniestro_viales.columns

Index(['Id', 'Rol', 'Sexo', 'Edad', 'Fecha_fallecimiento', 'Rango_etario',
       'Numero_víctimas', 'Fecha', 'Año', 'Mes', 'Día', 'Hora completa',
       'Hora', 'Lugar_del_hecho', 'Tipo_de_calle', 'Calle', 'Cruce',
       'Dirección normalizada', 'Comuna', 'Pos x', 'Pos y', 'Participantes',
       'Victima', 'Acusado', 'Coordenada_x', 'Coordenada_y', 'Barrios',
       'Día_Semana'],
      dtype='object')

Ordeno las columnas para que el Df sea más legible y fácil de intepretar

In [80]:
Ordenado=["Id", "Fecha", "Año", "Mes", "Día","Día_Semana", "Hora completa", "Hora", "Numero_víctimas", "Participantes", "Rol", "Acusado", "Victima", "Sexo", "Edad", "Rango_etario", "Fecha_fallecimiento", "Lugar_del_hecho", "Tipo_de_calle", "Calle", "Cruce", "Dirección normalizada", "Comuna", "Barrios", "Coordenada_x", "Coordenada_y", "Pos x", "Pos y"]

Siniestro_viales_limpio = Siniestro_viales[Ordenado]


In [79]:
Siniestro_viales_limpio

Id      Fecha   Año  Mes  Día Día_Semana Hora completa  Hora  \
0    2016-0001 2016-01-01  2016    1    1    Viernes      04:00:00     4   
1    2016-0002 2016-01-02  2016    1    2     Sabado      01:15:00     1   
2    2016-0003 2016-01-03  2016    1    3    Domingo      07:00:00     7   
3    2016-0004 2016-01-10  2016    1   10    Domingo      00:00:00     0   
4    2016-0005 2016-01-21  2016    1   21     Jueves      05:20:00     5   
..         ...        ...   ...  ...  ...        ...           ...   ...   
712  2021-0092 2021-12-12  2021   12   12    Domingo      06:20:00     6   
713  2021-0093 2021-12-13  2021   12   13      Lunes      17:10:00    17   
714  2021-0094 2021-12-20  2021   12   20      Lunes      01:10:00     1   
715  2021-0095 2021-12-30  2021   12   30     Jueves      00:43:00     0   
716  2021-0096 2021-12-15  2021   12   15  Miercoles      10:30:00    10   

     Numero_víctimas Victima  ... Tipo_de_calle                         Calle  \
0                  1    MOTO  ...       AVENIDA              PIEDRA BUENA AV.   
1                  1    AUTO  ...      GRAL PAZ                PAZ, GRAL. AV.   
2                  1    MOTO  ...       AVENIDA                ENTRE RIOS AV.   
3                  1    MOTO  ...       AVENIDA                LARRAZABAL AV.   
4                  1    MOTO  ...       AVENIDA                  SAN JUAN AV.   
..               ...     ...  ...           ...                           ...   
712                1  PEATON  ...       AVENIDA                 RIVADAVIA AV.   
713                1    MOTO  ...       AVENIDA                   RIESTRA AV.   
714                1    MOTO  ...     AUTOPISTA  DELLEPIANE, LUIS, TTE. GRAL.   
715                1    MOTO  ...       AVENIDA                     GAONA AV.   
716                1    AUTO  ...       AVENIDA                PERON, EVA AV.   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
712                       PUEYRREDON AV.   
713                                  MOM   
714                          LACARRA AV.   
715                              TERRADA   
716                                  NaN   

                                 Dirección normalizada Comuna  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
712                     RIVADAVIA AV. y PUEYRREDON AV.      3   
713                                  RIESTRA AV. y MOM      7   
714         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
715                                GAONA AV. y TERRADA     11   
716                                PERON, EVA AV. 4071      9   

                                               Barrios   Coordenada_x  \
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI   98896.782384   
1              LINIERS - MATADEROS - PARQUE AVELLANEDA   95832.055711   
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  106684.290900   
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI   99840.652248   
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  106980.328279   
..                                                 ...            ...   
712                          BALVANERA - SAN CRISTOBAL  105258.353686   
713                          FLORES - PARQUE CHACABUCO  102728.600901   
714            LINIERS - MATADEROS - PARQUE AVELLANEDA   99624.297958   
715  VILLA DEL PARQUE - VILLA  DEVOTO - VILLA GRAL....   99116.454924   
716          

In [71]:
Siniestro_viales_limpio.loc[:, 'Semestre'] = Siniestro_viales_limpio['Año'].astype(str) + '-' + Siniestro_viales_limpio['Mes'].apply(lambda x: '1' if x <= 6 else '2')

Siniestro_viales_limpio

Id      Fecha   Año  Mes  Día Día_Semana Hora completa  Hora  \
0    2016-0001 2016-01-01  2016    1    1    Viernes      04:00:00     4   
1    2016-0002 2016-01-02  2016    1    2     Sabado      01:15:00     1   
2    2016-0003 2016-01-03  2016    1    3    Domingo      07:00:00     7   
3    2016-0004 2016-01-10  2016    1   10    Domingo      00:00:00     0   
4    2016-0005 2016-01-21  2016    1   21     Jueves      05:20:00     5   
..         ...        ...   ...  ...  ...        ...           ...   ...   
712  2021-0092 2021-12-12  2021   12   12    Domingo      06:20:00     6   
713  2021-0093 2021-12-13  2021   12   13      Lunes      17:10:00    17   
714  2021-0094 2021-12-20  2021   12   20      Lunes      01:10:00     1   
715  2021-0095 2021-12-30  2021   12   30     Jueves      00:43:00     0   
716  2021-0096 2021-12-15  2021   12   15  Miercoles      10:30:00    10   

     Numero_víctimas   Participantes  ...                         Calle  \
0                  1       MOTO-AUTO  ...              PIEDRA BUENA AV.   
1                  1  AUTO-PASAJEROS  ...                PAZ, GRAL. AV.   
2                  1       MOTO-AUTO  ...                ENTRE RIOS AV.   
3                  1         MOTO-SD  ...                LARRAZABAL AV.   
4                  1  MOTO-PASAJEROS  ...                  SAN JUAN AV.   
..               ...             ...  ...                           ...   
712                1     PEATON-AUTO  ...                 RIVADAVIA AV.   
713                1       MOTO-AUTO  ...                   RIESTRA AV.   
714                1       MOTO-AUTO  ...  DELLEPIANE, LUIS, TTE. GRAL.   
715                1     MOTO-CARGAS  ...                     GAONA AV.   
716                1     AUTO-CARGAS  ...                PERON, EVA AV.   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
712                       PUEYRREDON AV.   
713                                  MOM   
714                          LACARRA AV.   
715                              TERRADA   
716                                  NaN   

                                 Dirección normalizada Comuna  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
712                     RIVADAVIA AV. y PUEYRREDON AV.      3   
713                                  RIESTRA AV. y MOM      7   
714         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
715                                GAONA AV. y TERRADA     11   
716                                PERON, EVA AV. 4071      9   

                                               Barrios   Coordenada_x  \
0       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI   98896.782384   
1              LINIERS - MATADEROS - PARQUE AVELLANEDA   95832.055711   
2    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  106684.290900   
3       VILLA LUGANO - VILLA RIACHUELO - VILLA SOLDATI   99840.652248   
4    CONSTITUCION - MONTSERRAT - PUERTO MADERO - RE...  106980.328279   
..                                                 ...            ...   
712                          BALVANERA - SAN CRISTOBAL  105258.353686   
713                          FLORES - PARQUE CHACABUCO  102728.600901   
714            LINIERS - MATADEROS - PARQUE AVELLANEDA   99624.297958   
715  VILLA DEL PARQUE - VILLA  DEVOTO - VILLA GRAL....   99116.454924   
716            LINIERS - MATADEROS - PARQUE AVELLANEDA   99324.544640   

      Coord

Guardo el Df Siniestro_viales_limpio para ser utilizado posteriormente

In [72]:
Siniestro_viales_limpio.to_csv("../Datasets/Siniestro_viales_limpio.csv", index=False)